catroid/src/main/java/org/catrobat/catroid/ui/WebViewActivity.java

```diff
@@ -219,14 +219,18 @@ public void createProgressDialog(String mediaName) {
 		progressDialog.setProgress(0);
 		progressDialog.setMax(100);
 		progressDialog.setProgressNumberFormat(null);
-		progressDialog.show();
+		if (!isFinishing()) {
+			progressDialog.show();
+		}
 	}
 
 	public void updateProgressDialog(long progress) {
 		if (progress == 100) {
-			progressDialog.setProgress(progressDialog.getMax());
-			setResult(RESULT_OK, resultIntent);
-			progressDialog.dismiss();
+			if (progressDialog.isShowing() && !isFinishing()) {
```

Comment:
what if we completely delete the progressDialog. As a user you mostly never see the dialog anyway

Reply:
i thought about that.
in the long run, it would be probably better, BUT....
the problem that multiple downloads could be started by fast tapping two different files still persists, and the way the webview activity handles downloads it is not possible to quickfix this.

so the webview would have to be refactored or better written from scratch to deal with all of this properly. (for example, the cache file and result intent are created on download start and saved as members (!),and then used again when download finished, which is triggered not by the download service/task callback directly, but the progress update of the dialog to "100") :-P

for now i think it would be best to just try to prevent the crashes. (which is not guaranteed with this fix anyways, since i still cannot reproduce them)